site/_posts/2021-03-16-securely-implement-oauth-vuejs.md

```diff
@@ -1174,32 +1221,310 @@ methods: {
+Once we have sent `userData` to our server, we reset the `textarea` by setting  `userData` equal to `''`, since it is a two-way binding. To bind this function to the `submit` event we added the following to the `form` tag:
 
 ```vue
 <form @submit.prevent="update">
   //
 </form>
 ```
 
-> FUN FACT: Using `.prevent` stops the page from reloading whenever the Submit button is clicked. 
+Also, be sure and update your `App.vue` to account for the user-data.  Here is a snippet of what your `App.vue` should look like from the improvements in this last section.
+
+```vue
+import Greet from './Greeting';
+import Login from "./Login";
+import Update from "./Update";
+
+export default {
+  name: 'app',
+  components: {
+    Greet,
+    Login,
+    Update
+  },
+//...
+```
+
+> ***FUN FACT***: Using `.prevent` stops the page from reloading whenever the Submit button is clicked. 
+
+To summarize, here is a snapshot of the full code for `Update.vue` now:
+```vue
+<template>
+  <form @submit.prevent="update">
+  <textarea
+    v-model="userData"
+    placeholder="Update FusionAuth user data."
+  ></textarea>
+    <button type="submit" class="button">Submit</button>
+  </form>
+</template>
+<script>
+export default {
+  name: "Update",
+  data() {
+    return {
+      userData: "",
+    };
+  },
+  methods: {
+    update: function () {
+      fetch(`http://localhost:9000/set-user-data`, {
+        credentials: "include",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          userData: this.userData,
+        }),
+      }).catch((err) => {
+        console.log(err);
+      });
+      this.userData = ''
+    },
+  },
+}
+</script>
+<style>
+textarea {
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+}
+
+button {
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 5px;
+}
+</style>
+```
 
 Here is how our application looks now:
 
 {% include _image.liquid src="/assets/img/blogs/oauth-vuejs/signed-in-with-text-area.png" alt="The application when logged in with the text area for updating the data." class="img-fluid" figure=false %}
 
 Go to your Vue app and type some text in the `textarea` and click the Submit button. If you log in to the FusionAuth dashboard, you can now see the text you added is stored in FusionAuth. It is under the **User data** tab in your user account details.
 
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/custom-user-data.png" alt="custom user data" class="img-fluid top-cropped" figure=false %}
+
+## One final condition 
+
+If you have made it this far...Great!  You have a fully working example of a VueJS application using FusionAuth.  
+
+The remainder of this article will cover one final condition and improve our existing code.  
+
+### Authorization vs authentication 
+
+What happens if we create a user in FusionAuth but do not assign them to an application?  You may notice our ExpressJS server crashes.
```

Comment:
I just noticed you are using two spaces after a period. Let's standardize on one. If you could add that to the devdocsreadme under the blog section, that'd be great!